src/FormBuilder/resources/js/form-builder/src/hooks/useFieldNameValidator.ts

```diff
@@ -0,0 +1,104 @@
+import {useSelect} from '@wordpress/data';
+
+/**
+ * @since 0.1.0
+ *
+ * @returns {*}
+ */
+export const getFieldNameFrequency = (fieldName: string, fieldNames: string[]) => {
+    return fieldNames.filter((name) => name === fieldName).length;
+};
+
+export const hasFieldNameConflict = (fieldName: string, fieldNames: string[]) => {
+    return fieldNames.some((name) => name === fieldName);
+};
+
+/**
+ * @unreleased switch hyphens to underscores
+ * @since 0.1.0
+ *
+ * @returns {`${*}-${number|number}`}
+ */
+export const getFieldNameSuggestion = (name, names) => {
+    const [prefix] = name.split(/^(.*)_([0-9]*)$/g).filter(Boolean);
+    const similarFieldNames = names.filter((fieldName) => fieldName.startsWith(prefix));
+    const increments = similarFieldNames.flatMap((fieldName) => fieldName.split(/^.*_([0-9]*)$/g).filter(Number)) || [
+        0,
+    ];
+    const nextIncrement = increments.length ? Math.max(...increments) + 1 : 1;
+    return `${prefix}_${nextIncrement}`;
+};
+
+/**
+ * @since 0.1.0
+ */
+export const flattenBlocks = (block) => [block, ...block.innerBlocks.flatMap(flattenBlocks)];
+
+const builtInFieldNames = [
+    'amount',
+    'currency',
+    'gatewayId',
+    'email',
+    'company',
+    'name',
+    'firstName',
+    'lastName',
+    'honorific',
+    'billingAddress',
+    'country',
+    'address1',
+    'address2',
+    'city',
+    'state',
+    'zip',
+    'login',
+    'donation-summary',
+    'donationType',
+    'subscriptionFrequency',
+    'subscriptionInstallments',
+    'subscriptionPeriod',
+    'subscription_id',
+    'additional_email',
```

Comment:
@JasonTheAdams Actually one more to add `formId`

Reply:
Done. Resolved in 4a6ed741